src/main/proto/netflix/titus/titus_cluster_membership_api.proto

```diff
@@ -0,0 +1,205 @@
+syntax = "proto3";
+
+// Titus Membership Management API specification. This API provides access to Titus service operational state
+// (leader/non-leader, activation time, etc), as well as provides management operations.
+// The membership management API is intended to be used by internal tools, which should make the calls directly to the
+// desired service instance (not via proxy or load balancer). The API calls tunneled via gateway are executed
+// against any available service instance. All service nodes should have identical eventually consistent state.
+//
+
+package com.netflix.titus;
+
+import "google/protobuf/empty.proto";
+
+option java_multiple_files = true;
+option java_package = "com.netflix.titus.grpc.protogen";
+option java_outer_classname = "ClusterMembershipProto";
+
+option go_package = "titus";
+
+// ----------------------------------------------------------------------------
+// Titus supervisor data model.
+
+message ChangeRecord {
+
+    string instanceId = 1;
+
+    // (Required) Detailed code for being in a specific state.
+    string reasonCode = 2;
+
+    // (Optional) Textual description accompanying the 'reasonCode'.
+    string reasonMessage = 3;
+
+    // Time when a transition to a state happened.
+    uint64 timestamp = 4;
+}
+
+message ServicePort {
```

Comment:
I suggest changing this to `ServiceAddress` and moving ip information to here so you can separate out Instance information from how to communicate with it.

Reply:
I agree. I will change this accordingly.